File: Iteration2/code/CSE3063GRP12/src/test/UserRepositoryTest.java
package test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

import core.exceptions.UserNotFoundException;
import core.exceptions.WrongPasswordException;
import core.general_providers.SessionController;
import core.models.abstracts.User;
import core.models.concretes.Student;
import core.repositories.UserRepository;

public class UserRepositoryTest {

    @Test
    public void testLoginCheckSuccessful() throws UserNotFoundException, WrongPasswordException {
        UserRepository userRepository = new UserRepository();
        userRepository.loginCheck("o150121534", "123123");
        User user = SessionController.getInstance().getCurrentUser();
        assertNotNull(user);
    }

    @Test(expected = UserNotFoundException.class)
    public void testLoginCheckUserNotFound() throws UserNotFoundException, WrongPasswordException {
        UserRepository userRepository = new UserRepository();
        userRepository.loginCheck("o1501215", "123123");
    }

    @Test(expected = WrongPasswordException.class)
    public void testLoginCheckWrongPassword() throws UserNotFoundException, WrongPasswordException {
        UserRepository userRepository = new UserRepository();
        userRepository.loginCheck("o150121534", "1231234");
    }
}

File: Iteration3/code/CSE3063GRP12/src/UserRepositoryTest.py
import unittest
from core.repositories.UserRepository import UserRepository  
from core.general_providers.SessionController import SessionController  
from core.exceptions.UserNotFoundException import UserNotFoundException  
from core.exceptions.WrongPasswordException import WrongPasswordException  

class UserRepositoryTest(unittest.TestCase):

    def test_login_check_successful(self):
        userRepository = UserRepository()
        userRepository.login_check("o150120060", "150120060")
        user = SessionController.getInstance().get_current_user()
        self.assertIsNotNone(user)

    def test_login_check_user_not_found(self):
        userRepository = UserRepository()
        with self.assertRaises(UserNotFoundException):
            userRepository.login_check("o1501asdasd12", "150120060")

    def test_login_check_wrong_password(self):
        userRepository = UserRepository()
        with self.assertRaises(WrongPasswordException):
            userRepository.login_check("o150120060", "5151515")

if __name__ == '__main__':
    unittest.main()
